extensions/ql-vscode/test/vscode-tests/no-workspace/model-editor/method-modeling/method-modeling-view-provider.test.ts

```diff
@@ -0,0 +1,212 @@
+import type {
+  Uri,
+  Webview,
+  WebviewView,
+  WebviewViewResolveContext,
+} from "vscode";
+import { CancellationTokenSource, EventEmitter } from "vscode";
+import type { ModelConfigListener } from "../../../../../src/config";
+import { MethodModelingViewProvider } from "../../../../../src/model-editor/method-modeling/method-modeling-view-provider";
+import { createMockApp } from "../../../../__mocks__/appMock";
+import { createMockModelingEvents } from "../../../../__mocks__/model-editor/modelingEventsMock";
+import { createMockModelingStore } from "../../../../__mocks__/model-editor/modelingStoreMock";
+import { mockedObject } from "../../../../mocked-object";
+import type { FromMethodModelingMessage } from "../../../../../src/common/interface-types";
+import { DisposableObject } from "../../../../../src/common/disposable-object";
+import type { ModelingEvents } from "../../../../../src/model-editor/modeling-events";
+import type {
+  DbModelingState,
+  SelectedMethodDetails,
+} from "../../../../../src/model-editor/modeling-store";
+import { mockDatabaseItem } from "../../../utils/mocking.helpers";
+import {
+  createMethod,
+  createUsage,
+} from "../../../../factories/model-editor/method-factories";
+
+describe("method modeling view provider", () => {
+  // Modeling store
+  let getStateForActiveDb: jest.Mock<DbModelingState | undefined, []>;
+  let getSelectedMethodDetails: jest.Mock<
+    SelectedMethodDetails | undefined,
+    []
+  >;
```

Comment:
Would it make sense to use the function definitions here rather than manually specifying the arguments and return types?

```suggestion
  let getStateForActiveDb: jest.MockedFunction<
    ModelingStore["getStateForActiveDb"]
  >;
  let getSelectedMethodDetails: jest.MockedFunction<
    ModelingStore["getSelectedMethodDetails"]
  >;
```

Reply:
Nice. I wasn't aware of `jest.MockedFunction` but that is a lot cleaner.